src/main/java/graphql/annotations/TypeFunction.java

```diff
@@ -17,10 +17,59 @@
 import graphql.schema.GraphQLType;
 
 import java.lang.reflect.AnnotatedType;
-import java.util.Collection;
 import java.util.function.BiFunction;
 
+/**
+ * A GraphQLType builder for java types.
+ */
 public interface TypeFunction extends BiFunction<Class<?>, AnnotatedType, GraphQLType> {
+    /**
+     * @deprecated Override {@link #buildType(String, Class, AnnotatedType)} as an alternative.
+     * This interface will not extend {@link BiFunction} in a future release.
+     */
+    @Deprecated
+    default GraphQLType apply(Class<?> aClass, AnnotatedType annotatedType) {
```

Comment:
It's a disruptive change, but in the end we won't have a BiFunction as the type builder. I think making this a breaking change with a major version bump is a suitable choice.